### PR TITLE
Sidebar

### PR DESCRIPTION
**Pull request by ckj**
Sidebar screenshot
#### This PR:
- Kills global header in favor of a sidebar
- Fixed layout -> Fluid Layout
- Reduces chrome in an attempt to increase information density.
- Gives a user quick access to items they've bookmarked, been assigned, or have viewed recently
- Uses @cameronmcefee's monochromatic purple palette as a base for all text and borders
#### Remaining todos:
- [x] Rename `components/header/index.jsx` to `components/sidebar/index.jsx`
- [x] Update django templates to use sidebar component
- [x] Render assigned, bookmarked, and history in sidebar panels
- [x] Finish org switcher
- [x] Add notch for active sidebar item
- [x] Make sidebar responsive
- [x] Fix account view
- [x] Fix sudo view
- [x] Update shared group view
- [x] Update login / auth pages
- [x] Fix clickability issue w/ new project [screenshot](https://www.dropbox.com/s/bf6wg9pb4idm84g/Screenshot%202016-09-12%2022.42.35.png?dl=0)
- [x] Make short ids lowercase
- [x] Sort out permissions stuff in django templates (@benvinegar)
- [x] Use StatusPage.io data instead of mocks
- [x] Use letter avatars for org avatar until we support org image avatars
- [x] Update letter avatar colors to match brand colors
- [x] Fix routing error that occurs after clicking on a 'Recently Viewed' item
- [x] Add home icon and remove fancy hover stuff on org selector
- [x] Fix mobile user action section in header
- [x] Fix link hover color in alerts in `message-container`
- [x] Fix 'js out of date' alert on issue detail page
- [x] Center and remove border from sidebar loading state
- [x] Remove border from sidebar results and make result item background white
- [x] Fix back arrow placement on Add Member view [screenshot](https://cl.ly/3w3V3J1P1Y2L)
- [x] Home button doesn't work on non-react pages
- [x] Tests
- [ ] QA
#### Possible follow ups:
- [ ] Kill org dashboard
- [ ] Force a [get configured flow](https://cl.ly/242c1r0v0339/get-configured.png)
- [ ] Wire up Subscribe to project button

cc @getsentry/ui

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4025)

### CODE REVIEW

File: src/sentry/static/sentry/app/components/sidebarPanel.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          <h2>{this.props.title}</h2>
+        </div>
+        <div className="sidebar-panel-body">
+          {!this.props.children &&
+            <div className="sidebar-panel-items">
+              <SidebarPanelItem />
+            </div>
+          }
+          {this.props.children &&
+            this.props.children
```
Comment (macqueen) [inline src/sentry/static/sentry/app/components/sidebarPanel.jsx]: errrr, is this really necessary? i think `{this.props.children}` is the same, isn't it?

File: src/sentry/static/sentry/app/components/todos.jsx [hunk 2]
```diff
@@ -284,21 +266,8 @@ const TodoList = React.createClass({
...
-    if (this.state.seeAll) {
-      next_tasks = this.state.tasks;
-    } else {
-      next_tasks = this.state.tasks.filter( (task) => {
-        if (task.status != 'complete' && task.status != 'skipped') {
-          return task;
-        }
-      }).slice(0,3);
-    }
+    let next_tasks = this.state.tasks;
```
Comment (macqueen) [inline src/sentry/static/sentry/app/components/todos.jsx]: i know it was like this already, but why not `nextTasks`?

File: src/sentry/static/sentry/app/components/userLetterAvatar.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    let user = this.props.user;
+    return user.email || user.username || user.id || user.ip_address;
+  },
+
+  getDisplayName() {
+    let user = this.props.user;
+    return user.name || user.email || user.username || '';
+  },
+
+
```
Comment (macqueen) [inline src/sentry/static/sentry/app/components/userLetterAvatar.jsx]: extra newline

**Review by macqueen — COMMENTED** [review]
reviewed ~ half of this. will review the rest after interviews

File: src/sentry/static/sentry/app/actionCreators/incidents.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import ConfigStore from '../stores/configStore';
+import IncidentActions from '../actions/incidentActions';
+import $ from 'jquery';
+
+function getIncidentsFromIncidentResponse(incidents) {
+  if (incidents === null || incidents.length == 0) {
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/actionCreators/incidents.jsx]: Mmm ... seems like a wash, personally – people often trip on mismatched parens.

File: src/sentry/static/sentry/app/components/letterAvatar.jsx [hunk 5]
```diff
@@ -7,29 +7,31 @@ import React from 'react';
 
 const LetterAvatar = React.createClass({
   propTypes: {
-    user: React.PropTypes.object.isRequired
+    identifier: React.PropTypes.string.isRequired,
+    displayName: React.PropTypes.string.isRequired
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/letterAvatar.jsx]: Alas, we do `this.props.displayName.trim()` so it is required.

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+  renderRequiredActions() {
+    // TODO: investigate if this is seriously deprecated
+    let org = this.getOrganization();
+    let requiredAction = org && getFirstRequiredAdminAction(org);
+    let actionMessage;
+
+    if (org && requiredAction !== null) {
+      let slugId = requiredAction.ID.toLowerCase().replace(/_/g, '-');
+      let url = `/organizations/${org.slug}/actions/${slugId}/`;
+      actionMessage = (
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: Changed.

Also – this is _probably_ deprecated code and I was going to  investigate removing afterwards.

File: src/sentry/static/sentry/app/components/sidebarPanel.jsx (hunk 1 above)
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/sidebarPanel.jsx]: Fixed.

File: src/sentry/static/sentry/app/components/sidebar/organizationSelector.jsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    return <Link className={className} to={`/${org.slug}/`}>{child}</Link>;
+  },
+
+  render() {
+    // TODO: what to do in single org state?
+    // let isSingleOrg = ConfigStore.get('singleOrganization');
+    let activeOrg = this.props.organization;
+
+    // if (isSingleOrg || !activeOrg) {
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/sidebar/organizationSelector.jsx]: It only affects on-premise users, but it should be implemented. Tackling.

**Comment by ckj** [comment]
Todos from internal feedback:

- [ ] SSO login needs wallpaper/flag branding
- [ ] Reduce bottom margins [here](
https://www.dropbox.com/s/0a4gd1lp3xig7bi/Screenshot%202016-10-04%2015.01.30.png?dl=0)
- [ ] User Feedback link doesn't show on project settings (server views)
- [ ] No "modal" pages should show sidebar

**Pull request merged**